app/main/posts/modify/post-media.directive.js

```diff
@@ -16,39 +16,79 @@ function (
     return {
         restrict: 'E',
         replace: true,
-        require: '^^form',
+        require: 'ngModel',
         scope: {
-            mediaId: '=',
             media: '=',
             name: '@',
-            mediaHasCaption: '='
+            mediaHasCaption: '<'
         },
         template: require('./media.html'),
-        link: function ($scope, element, attr, formCtrl) {
-            if ($scope.mediaId) {
-                MediaEndpoint.get({id: $scope.mediaId}).$promise.then(function (media) {
-                    $scope.media = media;
-                    // Set initial media state
-                    $scope.media.changed = false;
-                });
-            } else {
-                // Initialize media object
-                $scope.media = {file: null, caption: null, dataURI: null, changed: false};
+        link: function ($scope, element, attr, ngModel) {
+            // Initialize media object
+            $scope.media = { id: null, file: null, caption: '', dataURI: null, changed: false };
+            $scope.mediaId = null;
+            $scope.showAdd = showAdd;
+            $scope.showReplace = showReplace;
+            $scope.showDelete = showDelete;
+            $scope.deleteMedia = deleteMedia;
+
+            activate();
+
+            function activate() {
+                renderViewValue();
+
+                // Watch for media changes
```

Comment:
Not the ideal way to detect changes, but it works. At some point we should refactor `file-upload` to accept `onChange` callbacks or similar.

Reply:
if you bind onchange on the  `element', it knows that it changed, It's just that the value is 'changed' and not an actual useful value. Obviously, still not ideal, just throwing it here in case it's something you wanted to do instead of watching over the media.changed. Not saying you `should` do  that because the benefit is not really that much .